WhisperAPI/WhisperAPI.Tests/Unit/SuggestionsControllerTest.cs

```diff
@@ -80,6 +93,32 @@ public void When_receive_valid_searchQuery_then_return_Ok_request(int validQuery
             suggestion.SuggestedDocuments.Should().BeEquivalentTo(this.GetListOfDocuments());
         }
 
+        [Test]
+        [TestCase(0)]
+        [TestCase(1)]
+        public void When_receive_invalid_or_null_selectQuery_then_return_bad_request(int invalidQueryIndex)
+        {
+            this._suggestionServiceMock = new Mock<ISuggestionsService>();
+            this._suggestionServiceMock
+                .Setup(x => x.UpdateContextWithSelectedSuggestion(It.IsAny<ConversationContext>(), new Guid("c21d07d5-fd5a-42ab-ac2c-2ef6101e58d1")))
+                .Returns(false);
+
+            this._suggestionController = new SuggestionsController(this._suggestionServiceMock.Object, null);
+            this._suggestionController.SelectSuggestion(this._invalidSelectQueryList[invalidQueryIndex]).Should().BeEquivalentTo(new BadRequestResult());
+        }
+
+        [Test]
+        [TestCase(0)]
+        public void When_receive_valid_selectQuery_then_return_Ok_request(int validQueryIndex)
+        {
+            this._suggestionServiceMock = new Mock<ISuggestionsService>();
+            this._suggestionServiceMock
+                .Setup(x => x.UpdateContextWithSelectedSuggestion(It.IsAny<ConversationContext>(), this._validSelectQueryList[validQueryIndex].Id.Value))
+                .Returns(true);
+            this._suggestionController = new SuggestionsController(this._suggestionServiceMock.Object, null);
+            this._suggestionController.SelectSuggestion(this._validSelectQueryList[validQueryIndex]).Should().BeEquivalentTo(new OkResult());
```

Comment:
I think that you are missing this line here :
"this._suggestionController.OnActionExecuting(this.GetActionExecutingContext(validQueryIndex));"
Because right now the OnActionExecuting of context controller is not called.

AND you can't use "GetActionExecutingContext" as is, because it's set up to 
".Setup(x => x.ActionArguments["searchQuery"])" but you want "selectQuery" instead